CODING_STANDARD.md

```diff
@@ -59,12 +59,13 @@ Formatting is enforced using clang-format. For more information about this, see
   ```
   Note that the `\file` tag must be immediately followed by a newline in order
   for Doxygen to relate the comment to the current file.
-- Each function should be preceded by a Doxygen comment describing that
-  function. The format should match the [LLVM
+- Each class, member variable and function should be preceded by a Doxygen
+  comment describing it when it is not immediately obvious what it does. The
+  format should match the [LLVM
```

Comment:
Maybe we also need an explicit note that {class,member,function} names must make clear their purpose?

Reply:
I've added a commit doing this, but in the _Naming_ section below